File: packages/extract/test/fixtures/polyglot/r/job.R
library(R6)

Job <- R6Class("Job",
  public = list(
    state = "queued",
    start = function() {
      if (self$state != "queued") return(invisible(self))
      self$state <- "running"
    },
    finish = function() {
      if (self$state == "running") self$state <- "done"
    },
    fail = function() {
      self$state <- "failed"
    },
    describe = function() {
      switch(self$state, queued = "waiting", running = "busy", retrying = "again")
    }
  )
)

read_first <- function(path) {
  con <- file(path, "r")
  line <- readLines(con, n = 1)
  close(con)
  line
}

File: packages/extract/test/fixtures/polyglot/ruby/job.rb
class Job
  def initialize
    @state = :queued
  end

  def start
    return unless @state == :queued
    @state = :running
  end

  def finish
    @state = :done if @state == :running
  end

  def fail
    @state = :failed
  end

  def describe
    case @state
    when :queued then "waiting"
    when :running then "busy"
    when :retrying then "again"
    end
  end
end
